src/index.js

```diff
@@ -157,6 +157,10 @@ class UglifyJsPlugin {
       uglifyOptions.output = uglifyOptions.output || {};
     }
 
+    if (typeof this.options.sourceMap === 'undefined' && ['sourcemap', 'source-map', 'hidden-source-map'].indexOf(compiler.options.devtool) > -1) {
```

Comment:
well, honestly I'm not certain of the history of this option.. I was just being careful to not break it's current usage.

That said, i see your point and im alright with removing the exposing of this option all together. Up to u.